src/components/admin/SearchableSelector.jsx

```diff
@@ -0,0 +1,125 @@
+/* eslint react/jsx-no-bind: 0 */
+
+import React from 'react';
+import update from 'react-addons-update';
+import _ from 'lodash';
+
+import BaseComponent from 'lib/BaseComponent';
+import SearchBar from 'components/admin/SearchBar';
+
+
+// material-ui
+import Chip from 'material-ui/Chip';
+
+class ObjectChip extends BaseComponent {
+  constructor(props) {
+    super(props);
+    this.onClick = this.onClick.bind(this);
+  }
+
+  onClick() {
+    this.props.onDelete(this.props.id);
+  }
+
+  render() {
+    return (
+      <Chip
+        onRequestDelete={this.onClick}
+        style={this.props.style}
+      >
+        {this.props.children}
+      </Chip>
+    );
+  }
+}
+
+export default class EditObjectsForm extends BaseComponent {
+  constructor(props) {
+    super(props);
+    this.safeSetState({
+      addObjectValue: '',
+      objectAutocomplete: [],
+    });
+    this.handleDelete = this.handleDelete.bind(this);
+    this.handleClickAdd = this.handleClickAdd.bind(this);
+  }
+
+  componentDidUpdate(prevProps) {
+    if (!this.props.disabled &&
+      this.props.objects !== prevProps.objects) {
+      this.props.onChange();
+    }
+  }
+
+  handleClickAdd(object) {
+    // disable this if saving
+    if (this.props.disabled) return;
+
+    const id = object.id;
+    const name = object.name;
+    const alreadyAdded = this.props.objects.find(baseObject => baseObject.id === id) !== undefined;
+
+    if (alreadyAdded) {
+      window.alert('Already added');
+      return;
+    }
+    const newObjects = update(this.props.objects, { $push: [{ id, name }] });
+    this.props.onUpdate(newObjects);
+  }
+
+  handleDelete(id) {
+    // disabled this if saving
+    if (this.props.disabled) return;
+
+    const index = this.props.objects.findIndex(object => object.id === id);
+    if (index === -1) {
+      throw new Error('The author you are trying to delete cannot be found');
```

Comment:
I also really want to modularize out our error logging, maybe I'll also just let you merge this in right now, but would be great to call a `logError` function or something like that here which I can then later hook into Sentry or any other kind of logging system we want to

Reply:
I'm going to leave this for later as I'm currently blocking Fred.

Reply:
Yep agree